diff --git a/src/Subfission/Cas/CasManager.php b/src/Subfission/Cas/CasManager.php
index c07253a..5a36203 100644
--- a/src/Subfission/Cas/CasManager.php
+++ b/src/Subfission/Cas/CasManager.php
@@ -60,7 +60,12 @@ class CasManager
 			phpCAS::setFixedServiceURL($this->config['cas_redirect_path']);
 		}
 
-		phpCAS::setServerLogoutURL($this->config['cas_logout_url']);
+        $serverBaseURL = 'https://' . $this->config['cas_hostname'] . $this->config['cas_uri']
+            . '/' . $this->config['cas_service_type'] . $this->config['cas_service_validate_query'];
+
+        phpCAS::setServerServiceValidateURL($serverBaseURL);
+
+        phpCAS::setServerLogoutURL($this->config['cas_logout_url']);
 
 		if ($this->config['cas_masquerade']) {
 			$this->_masquerading = true;
